dev/Tiltfile

```diff
@@ -569,3 +580,54 @@ k8s_resource(
     new_name='everest-server-deploy',
     labels=["everest"]
 )
+
+#################################
+############ Backup Storages ##########
+#################################
+def getBackupStorageRequestBody(storage_name):
+    return encode_json(
+        {
+            'name': storage_name,
+            'type': "s3",
+            'bucketName': '{0}-bucket'.format(storage_name),
+            'description': storage_name,
+            'url': "https://minio.minio.svc.cluster.local",
+            'region': "us-east-1",
+            'accessKey': "minioadmin",
+            'secretKey': "minioadmin",
+            'verifyTLS':  True,
+            'forcePathStyle': False,
+        }
+    )
+
+if need_backup_storage:
+    # Deploy minIO as backup storage
+    print('Deploying minIO as backup storage')
+    k8s_yaml('%s/.github/minio.conf.yaml' % backend_dir)
+    k8s_resource(
+        workload='minio',
+        objects=[
+            'minio:Namespace',
+            'minio-data:PersistentVolumeClaim:minio',
+            'minio-cert:Secret:minio'
+        ],
+        new_name='minIO-deploy',
+        labels=["backup-storage"]
+    )
+
+    admin_credentials = {'username': 'admin', 'password': UI_ADMIN_PASSWORD}
+    api_token_cmd='API_TOKEN=$(curl -f -v -L -s -H "Content-Type: application/json" -d \'{0}\' localhost:8080/v1/session | jq -r .token)'.format(encode_json(admin_credentials))
+    backup_storage_cmd_tmpl='curl -f -v -L -H "Content-Type: application/json" -H "Authorization: Bearer $API_TOKEN" -d \'{0}\' localhost:8080/v1/namespaces/{1}/backup-storages'
+    for namespace in config_yaml['namespaces']:
+        ns_name = namespace['name']
+        for storage_name in namespace.get('backupStorages', []):
+            request_body=getBackupStorageRequestBody(storage_name)
+            local_resource(
+                'bs:{0}:{1}'.format(ns_name, storage_name),
+                '{0} && {1}'.format(api_token_cmd, backup_storage_cmd_tmpl.format(request_body, ns_name)),
+                resource_deps = [
+                    'minIO-deploy',
+                    'everest-server-deploy',
+                ],
+                labels=["backup-storage"]
+            )
```

Comment:
Can't we do this the cloud-native way and create the BackupStorage CRs and respectively Secrets instead of using the API?
Relying on the API doesn't feel right. If you are testing some changes to the API this can fail. Also, this is not really idempotent, subsequent calls to this will have the curl command fail. Handling the k8s resources directly sounds better to me.

Reply:
Hmmm 🤔 I see 
But in this case do we really need to set the owner ref in the secret? That is only useful for garbage collection, so that when a BS is deleted the accompanying secret is also deleted. In this case since tilt would be handling both, it would also delete both when tearing down. So maybe we can avoid setting the owner ref in the secret 🤷

Reply:
Done